### PR TITLE
[Compute] vm create: New parameter --count

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

`az vm create --count`, create multiple VMs in one command.

**Testing Guide**

```
az vm create -g fy -n vm --image centos --count 3
```

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -345,6 +345,7 @@ def load_arguments(self, _):
                    help='Mode of in-guest patching to IaaS virtual machine. Possible values are: Manual - You  control the application of patches to a virtual machine. You do this by applying patches manually inside the VM. In this mode, automatic updates are disabled; the paramater --enable-auto-update must be false. AutomaticByOS - The virtual machine will automatically be updated by the OS. The parameter --enable-auto-update must be true. AutomaticByPlatform - the virtual machine will automatically updated by the OS. The parameter --enable-agent and --enable-auto-update must be true')
         c.argument('ssh_key_name', help='Use it as public key in virtual machine. It should be an existing SSH key resource in Azure.')
         c.argument('enable_hotpatching', arg_type=get_three_state_flag(), help='Patch VMs without requiring a reboot. --enable-agent must be set and --patch-mode must be set to AutomaticByPlatform', min_api='2020-12-01')
+        c.argument('count', type=int, help='How many virtual machines you want to create.')
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:348]:
```suggestion
        c.argument('count', type=int, help='Count of virtual machines you want to create. Can only be used with --name and --resource-group-name.')
```

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 2]
```diff
@@ -255,20 +262,26 @@ def build_vm_resource(  # pylint: disable=too-many-locals, too-many-statements
...
+        enable_hotpatching=None, count=None):
 
     os_caching = disk_info['os'].get('caching')
 
     def _build_os_profile():
 
         special_chars = '`~!@#$%^&*()=+_[]{}\\|;:\'\",<>/?'
 
+        # 2 is used to avoid shadow names
+        computer_name2 = computer_name or ''.join(filter(lambda x: x not in special_chars, name))
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:274]: how about name it better such as _computer_name

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py [hunk 3]
```diff
@@ -472,6 +485,12 @@ def _build_storage_profile():
     }
     if zone:
         vm['zones'] = zone
+    if count:
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:488]: any defensive check on --count not able to co-present with other subresource params?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:501]: I plan to add checks in validator.

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py (hunk 2 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_template_builder.py]: fixed

File: src/azure-cli/azure/cli/command_modules/vm/_template_builder.py (hunk 3 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_template_builder.py:521]: added

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py]: updated

**Review by yungezz — APPROVED** [review]

**Pull request merged**